packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialog.js

```diff
@@ -127,39 +127,56 @@ const mapDispatchToProps = dispatch => ({
   onCloseOverlay: () => dispatch(closeEnlargedDialog()),
   onSetDateRange: (startDate, endDate) =>
     dispatch(setEnlargedDashboardDateRange(startDate, endDate)),
-  onSetDrillDownDateRange: (startDate, endDate) =>
-    dispatch(setDrillDownDateRange(startDate, endDate)),
+  onSetDrillDownDateRange: (startDate, endDate, currentLevel) =>
+    dispatch(setDrillDownDateRange(startDate, endDate, currentLevel)),
   dispatch,
 });
 
 const mergeProps = (stateProps, { dispatch, ...dispatchProps }, ownProps) => ({
   ...stateProps,
   ...dispatchProps,
   ...ownProps,
+  // When drilling down, we use the date range of the first layer as default but respect the granularity of the second layer
```

Comment:
I'm missing the bit about it respecting the granularity of the second layer 🤔 

There are a few different cases that we could possibly handle (of different granularities between first and second layer), but the only really important one is that if the second layer has no periodGranularity set, it doesn't constrain the fetch by any date ranges. Is this working?

Reply:
The granularity gets passed as part of the `drillDownConfig`.

I have added the logic inline to handle the case of no date being set (I think that is the best, most obvious place for it). As you say there could be other cases / exceptions that we could add but they seem less clear cut.